src/common/imageio.c

```diff
@@ -228,7 +228,7 @@ gboolean dt_imageio_has_mono_preview(const char *filename)
 {
   dt_colorspaces_color_profile_type_t color_space;
   uint8_t *tmp = NULL;
-  int32_t thumb_width, thumb_height = 0;
+  int32_t thumb_width = 0, thumb_height = 0;
```

Comment:
Without explicit initialization, `thumb_width` may be used uninitialized in the error handling code after the `cleanup` label.